projects/npm-tools/packages/npm-scripts/src/scripts/build.js

```diff
@@ -64,6 +58,18 @@ if (!BUILD_CONFIG) {
  * `minify()` is run unless `NODE_ENV` is `development`.
  */
 module.exports = async function (...args) {
+	const config = getMergedConfig('npmscripts');
+
+	createTempFile('npmscripts.config.json', JSON.stringify(config, null, 2), {
+		autoDelete: false,
```

Comment:
I gather the plan is to remove this once we're confident it works, right? In the past we worked pretty hard to keep untracked/dirty files out of the worktree.

Reply:
This is not generated in the source tree, but in the temporary directory, which is configurable and (in our case) is placed inside the build folder. Remember we had to move everything out of the source tree because of the `watch` feature.

The reason we need these files is that, for any diagnostic, it is quite useful to know what the tools are really seeing. Think of it as the report file in the bundler, which helped a lot with that. Also, for learning purposes, some users may find these files useful.

Otherwise, with all the complex inheritance we have in place, it's nearly impossible to know what is being fed to webpack, the federation plugin, and so on, without spending a good amount of time looking at different places.

That said, I'm sure we will end up doing this some other way, once we settle all this stuff... Perhaps a report file or some kind of advanced diagnostic tool that can be used by developers to know what's going on in their builds. Maybe a log level parameter could work too...

Reply:
> This is not generated in the source tree, but in the temporary directory, which is configurable and (in our case) is placed inside the build folder.

Bah, and you probably told me this the last time I asked about it too. 🤦‍♂️

![sorryz](https://user-images.githubusercontent.com/7074/109668005-03c9a680-7b71-11eb-8974-7b32639aeea8.gif)